fleet/fleet.go

```diff
@@ -131,6 +132,15 @@ type Fleet interface {
 func NewFleet(config Config) (Fleet, error) {
 	var trans http.RoundTripper
 
+	tunneling := config.Tunnel.Tunnel != ""
+	// If a tunnel in provided we need to overwrite the http.Transport.Dial function
+	// to use the tunnel
+	if tunneling {
+		trans = &http.Transport{
+			Dial: config.Tunnel.NewDialFunc(config),
+		}
+	}
```

Comment:
I would say we don't need to execute the whole switch logic in case we tunnel through ssh. Then line 167 can be dropped and the code flow is more straight forward. What do you think?